python/doc/examples/numerical_methods/optimization_bonmin.ipynb

```diff
@@ -0,0 +1,165 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Optimization using bonmin"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example we are going to explore mixed-integer non linear problems optimization using OpenTURNS' [bonmin](https://www.coin-or.org/Bonmin/index.html) interface."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from __future__ import print_function\n",
+    "import openturns as ot"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "B-BB\n",
+      "B-OA\n",
+      "B-QG\n",
+      "B-Hyb\n",
+      "B-iFP\n"
+     ]
+    }
+   ],
+   "source": [
+    "# List available algorithms\n",
+    "for algo in ot.Bonmin.GetAlgorithmNames():\n",
+    "    print(algo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Details and references on bonmin algorithms are available [here](https://projects.coin-or.org/Bonmin)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setting up and solving a simple problem\n",
+    "\n",
+    "The following example will demonstrate the use of bonmin \"BB\" algorithm to solve the following problem:\n",
+    "\n",
+    "$$ \\min - x_0 - x_1 - x_2 $$\n",
+    "    \n",
+    "such that:\n",
+    "    \n",
+    "$$\\begin{array}{l}\n",
+    "(x_1 - \\frac{1}{2})^2 + (x_2 - \\frac{1}{2})^2 \\leq \\frac{1}{4} \\\\\n",
```

Comment:
This equation does not match the code. The code is >= and the equation is <=.

Reply:
In the presentation of the problem, the inequality is stated in the most simple way, for the sake of readability. In the code, it is re-stated as h(x) >= 0 to match openturns' requirements.
However, the two forms are equivalent.
Same goes for the other inequality constraints.

Reply:
Ok, so to make the point, you might as well 
* first write the equations as simple as they can (as the present code),
* then write them in the format required by Bonmin. 
This way, the user can explicitly see how this requirement can be taken into account.

Reply:
OK, I added a comment after the presentation of the problem to explain how equality/inequality constraints have to be restated to match OT's requirements.